packages/driver/cypress/integration/commands/actions/type_events_spec.js

```diff
@@ -0,0 +1,770 @@
+          location: 0,
+          metaKey: false,
+          repeat: false,
+          shiftKey: false,
+          type: 'keydown',
+          which: 65, // deprecated but fired by chrome
+        })
+
+        done()
+      })
+
+      cy.get(':text:first').type('a')
+    })
+
+    it('receives keypress event', (done) => {
+      const $txt = cy.$$(':text:first')
+
+      $txt.on('keypress', (e) => {
+        expect(_.toPlainObject(e.originalEvent)).to.include({
+          altKey: false,
+          bubbles: true,
+          cancelable: true,
+          charCode: 97, // deprecated
+          ctrlKey: false,
+          detail: 0,
+          key: 'a',
+          code: 'KeyA',
+          keyCode: 97, // deprecated
+          location: 0,
+          metaKey: false,
+          repeat: false,
+          shiftKey: false,
+          type: 'keypress',
+          which: 97, // deprecated
+        })
+
+        done()
+      })
+
+      cy.get(':text:first').type('a')
+    })
+
+    it('receives keyup event', (done) => {
+      const $txt = cy.$$(':text:first')
+
+      $txt.on('keyup', (e) => {
+        expect(_.toPlainObject(e.originalEvent)).to.include({
+          altKey: false,
+          bubbles: true,
+          cancelable: true,
+          charCode: 0, // deprecated
+          ctrlKey: false,
+          detail: 0,
+          key: 'a',
+          code: 'KeyA',
+          keyCode: 65, // deprecated but fired by chrome always uppercase in the ASCII table
+          location: 0,
+          metaKey: false,
+          repeat: false,
+          shiftKey: false,
+          type: 'keyup',
+          view: cy.state('window'),
+          which: 65, // deprecated but fired by chrome
+        })
+        .not.have.property('inputType')
+
+        done()
+      })
+
+      cy.get(':text:first').type('a')
+    })
+
+    it('receives textInput event', (done) => {
+      const $txt = cy.$$(':text:first')
+
+      $txt[0].addEventListener('textInput', (e) => {
+        // FIXME: (firefox) firefox cannot access window objects else throw cross-origin error
+        expect(Object.prototype.toString.call(e.view)).eq('[object Window]')
+        e.view = null
+        expect(_.toPlainObject(e)).to.include({
+          bubbles: true,
+          cancelable: true,
+          data: 'a',
+          detail: 0,
+          type: 'textInput',
+          // view: cy.state('window'),
+          which: 0,
+        })
+
+        done()
+      })
+
+      cy.get(':text:first').type('a')
+    })
+
+    it('receives input event', (done) => {
+      const $txt = cy.$$(':text:first')
+
+      $txt.on('input', (e) => {
+        const obj = _.pick(e.originalEvent, 'bubbles', 'cancelable', 'type')
+
+        expect(obj).to.deep.eq({
+          bubbles: true,
+          cancelable: false,
+          type: 'input',
+        })
+
+        done()
+      })
+
+      cy.get(':text:first').type('a')
+    })
+
+    it('fires events in the correct order')
+
+    it('fires events for each key stroke')
```

Comment:
lets remove these empty tests or log an issue to back-fill these.

Reply:
I tracked the history of `type_spec.js` and found out that it was like this from the beginning(6faba005c7feffa7ca3a387351647ade8b32b553). IMHO, logging an issue and implementing these tests would be a better idea. But I'm not sure why it is left empty.

Reply:
Sounds like a good idea. Mind logging the issue & add TODO with link to these?

Reply:
I opened this issue at #20283.